WcaOnRails/app/javascript/edit-schedule/EditSchedule.jsx

```diff
@@ -0,0 +1,218 @@
+import React from 'react'
+import cn from 'classnames'
+import _ from 'lodash'
+import {
+  Alert,
+  Col,
+  Panel,
+  PanelGroup,
+  Row,
+} from 'react-bootstrap'
+
+import { rootRender, promiseSaveWcif } from 'edit-schedule'
+import { EditVenue } from './EditVenue'
+import { SchedulesEditor } from './SchedulesEditor'
+import { initElementsIds, newVenueId } from './utils'
+
+export const schedulesEditPanelSelector = "#schedules-edit-panel";
+
+export default class EditSchedule extends React.Component {
+  componentWillMount() {
+    this.setState({ savedScheduleWcif: _.cloneDeep(this.props.competitionInfo.scheduleWcif) });
+    initElementsIds(this.props.competitionInfo.scheduleWcif.venues);
+  }
+
+  save = e => {
+    let { competitionInfo } = this.props;
+    let wcif = {
+      id: competitionInfo.id,
+      schedule: competitionInfo.scheduleWcif,
+    };
+
+    this.setState({ saving: true });
+    promiseSaveWcif(wcif).then(response => {
+      return Promise.all([response, response.json()]);
+    }).then(([response, json]) => {
+      if(!response.ok) {
```

Comment:
This feels a bit weird to me. Can't you check `response.ok` in the first `then`, and then you don't have to use `Promise.all`? I'm thinking something like this:

```javascript
...then(response => {
  if(!response.ok) {
    throw ...;
  }
  return response.json();
}).then(json => {
  ...
}
```

Reply:
Ah, you're totally right, my bad!